protocol.txt

```diff
@@ -0,0 +1,185 @@
+            3.2.1. Server Command Listing .............................X
+            3.2.2. Server Command Arguments ...........................X
+        3.3. Client Commands ..........................................X
+            3.3.1. Response Format ....................................X
+            3.3.2. Client Command Listing .............................X
+            3.3.3. Client Command Arguments ...........................X
+
+1. Introduction
+
+    The R-Type project is an epitech project that aims to produce a
+    multiplayer game based on the R-Type game.
+
+    This project aims to introduce students to the intricate details of
+    building a networked multiplayer game while also focusing on the
+    package the game is delivered in. This includes but is not limited
+    to :
+    
+        - Crossplatform build system
+        - Limited user setup required
+        - Multithreaded server
+        - Inability to cheat using modified client
+
+2. TCP Protocol
+
+                             <TO BE WRITTEN>
+
+3. UDP Protocol
+
+    The UDP protocol will be the main protocol used to communicated the
+    behaviour of the game to clients during any given game.
+    
+    Since in this project the server is supposed to handle most of the
+    game logic this protocol will be centered around the ECS and
+    communicating it's state over the course of a given game.
+
+    NO UDP SERVER COMMAND will await any type of response.
+
+    3.1. Command Synthax
+
+        Commands follow the following syntax :
+
+                <CMD> <SP> <ARGS> <CRLF>
+        
+        Here is a definition of each part :
+
+        <CMD> -> This is the command name and will always be fully
+                 capitalized.
+
+        <SP> -> Ascii space character
+
+        <ARGS> -> These are the arguments for the command. The size of
+                  these args may vary even within the same command. For
+                  this purpose each argument will be separated by the
+                  ';' character.
+                  
+                  Any sub-argument within each argument will be
+                  separated by the ',' character.
+
+        <CRLF> -> Carriage Return Line Feed character. Every message
+                  will end in this character combination denoted '\r\n'
+                  in ascii.
+        
+        Here is an example of a valid server command :
+
+                             DEL_ENT 1 \r\n
+        
+    3.2. Server Commands
+
+        This section details the commands sent by the server to clients
+        along with their usage.
+
+        3.2.1 Server Command Listing
+
+            The following is a listing of all valid server commands :
+
+                    NEW_ENT       ->    Create entity
+                    DEL_ENT       ->    Delete entity
+                    ADD_COMP      ->    Add component
+                    DEL_COMP      ->    Delete component
+                    UPDATE_COMP   ->    Update component
```

Comment:
instead of UPDATE_COMP we could use UPDATE_ENT
similar to NEW_ENT except that you specify only the
components that are to be updated

Reply:
So like add a new command ? I feel like the ability to update a single component at the time would still be quite interesting to have.

Reply:
if that would be needed, then just UPDATE_ENT 342(Health(3))
                                                                                ^
                                                                            entity id